src/server/models/listItem/summary.helpers.ts

```diff
@@ -0,0 +1,134 @@
+import {Event, ListItem, ListItemEvent, Prisma, Status} from "@prisma/client";
+import {ActivityStatusViewModel} from "server/models/listItem/types";
+
+
+export const statusToActivityVM: Record<Status, ActivityStatusViewModel> = {
+  NEW: {
+    type: 'to_do',
+    text: 'Check new entry'
+  },
+  OUT_WITH_PROVIDER: {
+    type: 'out_with_provider',
+    text: 'Edits requested'
+  },
+  EDITED: {
+    text: 'Check edits',
+    type: 'to_do'
+  },
+  CHECK_ANNUAL_REVIEW: {
+    text: 'Check annual review',
+    type: 'to_do'
+  },
+  ANNUAL_REVIEW_OVERDUE: {
+    text: 'Annual review overdue',
+    type: 'out_with_provider'
+  },
+  PUBLISHED: {
+    text: 'No action needed',
+    type: 'no_action_needed'
+  },
+  UNPUBLISHED: {
+    text: 'Removed by post',
+    type: 'to_do',
+    colour: 'red',
+  },
+}
+
+
+export enum PUBLISHING_STATUS {
+  new = "new",
+  live = "live",
+  unpublished = "unpublished",
+  archived = "archived"
+}
+
+
+export type ListItemWithHistory = ListItem & {
+  history: Event[]
+}
+
+/**
+ * The prisma query must be called with
+ * history: {
+ *   orderBy: {
+ *     time: 'desc',
+ *   }
+ * },
+ * These functions do not do any additional sorting.
+ */
+export function newestEventOfTypeIndex(history: Event[], type: ListItemEvent): number {
+  return history.findIndex(event => event.type === type)
+}
+
+export function hasBeenUnpublishedSincePublishing(history: Event[]): boolean {
+  const newestPublishEvent = newestEventOfTypeIndex(history, ListItemEvent.PUBLISHED);
+  const newestUnpublishEvent = newestEventOfTypeIndex(history, ListItemEvent.UNPUBLISHED);
+  const hasUnpublishEvent = newestUnpublishEvent !== -1;
```

Comment:
What is `-1` in this case? That no event exists?

Reply:
Yea, findIndex returns -1 if it doesn’t find anything